modules/javafx.graphics/src/main/java/com/sun/glass/ui/mac/MacApplication.java

```diff
@@ -455,4 +460,23 @@ public Map<String, Class<?>> getPlatformKeys() {
             Map.entry("macOS.NSColor.systemYellowColor", Color.class)
         );
     }
+
+    private static final String SUPPRESS_AWT_WARNING_PROPERTY = "javafx.preferences.suppressAppleAwtWarning";
+    private static final String AWT_APPEARANCE_PROPERTY = "apple.awt.application.appearance";
+    private boolean checkSystemAppearance = !Boolean.getBoolean(SUPPRESS_AWT_WARNING_PROPERTY);
```

Comment:
This needs to be wrapped in a `doPrivileged` call, with a  `@SuppressWarnings` annotation, or it will fail to run with a security manager (which we still support as of now, even though it is deprecated).

See, for example, `PlatformImpl.java` for the pattern to use.